### ISSUE TITLE
Maninpasta minutes: BuildingKG#1

### ISSUE THREAD

**Issue by enridaga** [polifonia-project/stories#6]
To collect notes on the discussion in this group

**Comment by enridaga** [comment]
It would be useful to start collect a list of resources that the project members would like to include in the KG

**Comment by 85jesse** [comment]
The Dutch Network for Digital Heritage has developed a registry for datasets: https://github.com/netwerk-digitaal-erfgoed/register. This could provide an infrastructural component for Polifonia datasets as well. By registering them in this fashion the dataset-metadata can be queried, and various endpoints can be found.

**Comment by enridaga** [comment]
A catalogue of musical resources on the Web was developed a few years ago: https://musow.kmi.open.ac.uk/

**Comment by 85jesse** [comment]
For the project it will be relevant to create an overview of relevant datasets and their status (the degree to which they are published as semantic or linked data), but also their legal status.
Could be anything from data that needs to be scraped from html pages, to full RDF datasets. 
Then within Polifonia pipelines can be created (or use existing pipelines) to transform the data so that it can be used within the project.

**Comment by paolobonora** [comment]
Example of sources for the Carolina#1 CS about Perti, Giacomo Antonio:
1) Catalogue of Museo della Musica of Bologna: http://www.bibliotecamusica.it/cmbm/scripts/gaspari/libri.asp?ms=%27E%27&ms=%27M%27&ID=3589
2) Corago LOD: http://coragolod2.ing.unibo.it:8080/corago/resource/RESPONS/APCN00004400
3) Data from REPIM: 13 works from relational DB (we own)
4) Grove Music Online: https://doi.org/10.1093/gmo/9781561592630.article.21394

**Comment by enridaga** [comment]
Interested in the REPIM, "Repertorio Poesia in Musica"

**Comment by enridaga** [comment]
Obviously, there is a page on Wikipedia: https://en.wikipedia.org/wiki/Giacomo_Antonio_Perti

**Comment by paolobonora** [comment]
And we should also take into account the VIAF entry: http://viaf.org/viaf/19946155

**Comment by enridaga** [comment]
Take away message 1: we need a registry!

**Comment by paolobonora** [comment]
Another example: RISM (Répertoire International des Sources Musicales): https://opac.rism.info/metaopac/refineSearch.do;jsessionid=3DA5C69F3B0EFAE7E062CB0F2635B567.touch01?id=author_facet&methodToCall=filterSearch&subval=Perti%2CGiacomoAntonio

**Comment by enridaga** [comment]
Take away message 2: there is a huge diversity of formats / availability status / quality

**Comment by enridaga** [comment]
On the diversity of format, the OU is working on a new tool to SPARQL non-RDF resources:  https://github.com/SPARQL-Anything/sparql.anything

**Comment by paolobonora** [comment]
RISM is "available" also in RDF: https://opac.rism.info/id/rismid/454006820?format=rdf

**Comment by 85jesse** [comment]
The registry needs to be filled, which is open to all. But we would also like to bring datasets a step further towards full Linked Open Data publication (by setting up pipelines, settling on a dataformat, linking with various sources, etc), but there we need to focus our efforts. So we need to see which datasets need to be prioritised: are there any specific usecases within Polifonia that have a need for specific datasets?

**Comment by enridaga** [comment]
One challenge is that we cannot ask all data providers to commit to one ontological representation, which creates an interesting challenge in terms of developing an exploratory system for the KG

**Comment by 85jesse** [comment]
We also have to formulate requirements for datasets requirements in terms of performance (what kind of queries do they need to be able to handle? How long can it take before results are returned?)

**Comment by enridaga** [comment]
Maybe we need to pursue a mixed strategy, by leaving data to the provider but caching as soon as access is requested. A search facility will necessarily need to index all the data, though.

**Comment by 85jesse** [comment]
Datasets to be registered also include thesauri/vocabularies that are being used within the music-domain. Would also be interesting to know which vocabularies have been linked/aligned with other (public) datasources (e.g. Wikidata, Discogs, etc.). These vocabularies can act as linking layers in the Knowledge Graph.

**Comment by enridaga** [comment]
Maybe we can use GitHub as a repository for the registry, and include a JSON-(LD) file for each of the resources. The musoW web application can just expose data from there

**Comment by enridaga** [comment]
We should also consider using GitHub to host the actual datasets / linked data

**Comment by paolobonora** [comment]
We should define a basic process for a request of a new source within the KG.
Something like:
1) proposed from user
2) under analisys/triage
3) accepted and being aligned
4) added (and what has ben included)
5) rejected

**Comment by enridaga** [comment]
There is a set of key-questions about integrating resources into the knowledge graph: what do we integrate?

- Metadata about the resources (easy, mandatory)
- Schema elements / vocabularies
- Entity linking

**Comment by enridaga** [comment]
Are we expecting to copy the original resource and transform it with our vocabulary? Instead, are we asking data providers to commit to our representation? Something in between the two extremes?

**Comment by 85jesse** [comment]
Another possible piece of the puzzle for vocabularies is the 'Network of Terms', an application that allows you to search multiple vocabularies via a single API: https://github.com/netwerk-digitaal-erfgoed/network-of-terms-api 
This gives a good overview of vocabularies that are already out there that can be used. It's up to the collection holders to decide which vocabularie(s) they want to use.

**Comment by enridaga** [comment]
Work in progress query to generate schema.org descriptions from the current musoW catalogue:
```
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX schema: <http://schema.org/>
PREFIX dc: <http://purl.org/dc/elements/1.1/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
CONSTRUCT {
  ?item schema:identifier ?identifier ;
    	schema:license ?license ;
        schema:featureList ?feature
} 
FROM <http://data.open.ac.uk/context/musow>
WHERE {
VALUES (?item) {(<http://data.open.ac.uk/musow/79e8954f4b0bc004e3ed8e5ea91bf7b4>)} .
  ?item dct:identifier ?identifier ;
        <http://data.open.ac.uk/musow/ontology/access/type> ?charged ;
        <http://data.open.ac.uk/musow/ontology/situation/task> ?task ;
        <http://dbpedia.org/ontology/category> ?category ;
        <http://purl.org/dc/terms/accessRights> ?accessRights ;
        <http://purl.org/dc/terms/license> ?license ;
        <http://schema.org/featureList> ?feature
  .
}
```

**Comment by enridaga** [comment]
Shall we rename this issue into RegistryActivity?

**Comment by enridaga** [comment]
Shall we move this issue to the Registry repository?

**Comment by albertmeronyo** [comment]
Meeting 14-05-2021 on Sethus: @enridaga suggested adding MEI document support in SPARQL Anything to enable KG access/ingestion/creation

AP: presenting the idea (and maybe a prototype if time allows?) at the MEI WG meeting on 28-05-2021 would be great
